src/browser-lib/posthog/posthog.ts

```diff
@@ -22,22 +22,27 @@ export const posthogToAnalyticsServiceWithoutQueue = (
   name: "posthog",
   init: ({ apiKey, apiHost, uiHost }) =>
     new Promise((resolve) => {
-      client.init(apiKey, {
-        api_host: apiHost,
-        ui_host: uiHost,
-        debug: getBrowserConfig("releaseStage") !== "production",
-        loaded: () => {
-          const legacyAnonymousId = getLegacyAnonymousId();
-          if (legacyAnonymousId) {
-            client.register({
-              legacy_anonymous_id: legacyAnonymousId,
-            });
-          }
-          resolve(client.get_distinct_id());
-        },
-        disable_session_recording: true,
-        capture_pageview: false,
-      });
+      if (!client.__loaded) {
+        client.init(apiKey, {
+          api_host: apiHost,
+          ui_host: uiHost,
+          debug: getBrowserConfig("releaseStage") !== "production",
+          loaded: () => {
+            const legacyAnonymousId = getLegacyAnonymousId();
+            if (legacyAnonymousId) {
+              client.register({
+                legacy_anonymous_id: legacyAnonymousId,
+              });
+            }
+
+            resolve(client.get_distinct_id());
+          },
+          disable_session_recording: true,
+          capture_pageview: false,
+        });
+      } else {
+        resolve(client.get_distinct_id());
+      }
```

Comment:
Super subtle bug, but this promise would never resolve if Posthog had already been initialised. This happens in OWA currently because `AnalyticsProvider` is unmounted repeatedly on page load (not great).

This meant that `useAnalytics().posthogDistinctId` would always be `null`